src/_nebari/upgrade.py

```diff
@@ -507,6 +528,103 @@ def _version_specific_upgrade(
         return config
 
 
+class Upgrade_2023_9_1(UpgradeStep):
+    version = "2023.9.1"
+
+    def _version_specific_upgrade(
+        self, config, start_version, config_filename: Path, *args, **kwargs
+    ):
+        # Upgrading to 2023.9.1 is considered high-risk because it includes a major refacto
+        # to introduce the extension mechanism system.
+        rich.print("\n ⚠️  Warning ⚠️")
+        rich.print(
+            f"-> Nebari version [green]{self.version}[/green] includes a major refactor to introduce an extension mechanism that supports the development of third-party plugins."
+        )
+        rich.print(
+            "-> Data should be backed up before performing this upgrade ([green][link=https://www.nebari.dev/docs/how-tos/manual-backup]see docs[/link][/green])  The 'prevent_deploy' flag has been set in your config file and must be manually removed to deploy."
+        )
+        rich.print(
+            "-> Please also run the [green]rm -rf stages[/green] so that we can regenerate an updated set of Terraform scripts for your deployment."
+        )
+
+        # Setting the following flag will prevent deployment and display guidance to the user
+        # which they can override if they are happy they understand the situation.
+        config["prevent_deploy"] = True
+
+        # Nebari version 2023.9.1 upgrades JupyterHub to 3.1.  CDS Dashboards are only compatible with
+        # JupyterHub versions 1.X and so will be removed during upgrade.
+        rich.print("\n ⚠️  Deprecation Warning ⚠️")
+        rich.print(
+            f"-> CDS dashboards are no longer supported in Nebari version [green]{self.version}[/green] and will be uninstalled."
+        )
+        if config.get("cdsdashboards"):
+            rich.print("-> Removing cdsdashboards from config file.")
+            del config["cdsdashboards"]
+
+        # Kubernetes version check.  Minimum Kubernetes version is 1.26
+        # JupyterHub Helm chart 2.0.0 (app version 3.0.0) requires K8S Version >=1.23. (reference: https://z2jh.jupyter.org/en/stable/)
+
+        provider = config["provider"]
+        provider_config_block = get_provider_config_block_name(provider)
+
+        # Get current Kubernetes version if available in config.
+        current_version = config.get(provider_config_block, {}).get(
+            "kubernetes_version", "NA"
+        )
+
+        # Convert to decimal prefix
+        if provider in ["aws", "azure", "gcp", "do"]:
+            current_version = get_k8s_version_prefix(current_version)
+
+        # Try to convert known Kubernetes versions to float.
+        if not current_version == "NA":
+            try:
+                current_version = float(current_version)
+            except TypeError:
```

Comment:
This would most likely raise `ValueError` (?)

Reply:
I could have sworn when I tested this I got type error, but now that I'm trying again it's definitely ValueError - thanks!